src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -6,39 +6,100 @@ YARN_CACHE_DIR=/usr/local/share/yarn-cache
 if [ -d $YARN_CACHE_DIR ]
 then
 	echo
-    echo "Configuring Yarn cache folder ..."
+    echo "Configuring Yarn cache folder..."
     yarn config set cache-folder $YARN_CACHE_DIR
 fi
 
+zippedModulesFileName=node_modules.zip
+allModulesDirName=__oryx_all_node_modules
+prodModulesDirName=__oryx_prod_node_modules
+
+if [ "$ORYX_COPY_ONLY_PROD_MODULES_TO_OUTPUT" == "true" ]
+then
+	if [ -d $allModulesDirName ]
+	then
+		# Rename existing node_modules back to prod modules
+		mv node_modules $prodModulesDirName
+
+		# Rename the folder which has all the node modules to reuse for later builds to improve perf
+		mv $allModulesDirName node_modules
+	fi
+
+	hasProductionOnlyDependencies="{{ HasProductionOnlyDependencies }}"
+	if [ "$SOURCE_DIR" != "$DESTINATION_DIR" ] && [ "$hasProductionOnlyDependencies" == "true" ]
+	then
+		mkdir -p "$prodModulesDirName"
+		cd "$prodModulesDirName"
+
+		if [ -f "$SOURCE_DIR/package.json" ]; then
+			cp -f "$SOURCE_DIR/package.json" .
+		fi
+
+		if [ -f "$SOURCE_DIR/package-lock.json" ]; then
+			cp -f "$SOURCE_DIR/package-lock.json" .
+		fi
+
+		if [ -f "$SOURCE_DIR/yarn.lock" ]; then
+			cp -f "$SOURCE_DIR/yarn.lock" .
+		fi
+
+		echo
+		echo "Installing production dependencies in '$prodDependencies'..."
+		echo "Running '{{ ProductionOnlyPackageInstallCommand }}'..."
+		echo
+		{{ ProductionOnlyPackageInstallCommand }}
+
+		echo
+		echo "Copying production dependencies from '$prodDependencies' to '$SOURCE_DIR'..."
+		rsync -rtE --links node_modules "$SOURCE_DIR"
+	fi
+fi
+
+cd "$SOURCE_DIR"
+
 echo
-echo Installing packages ...
-echo
-echo "Running '{{ PackageInstallCommand }}' ..."
+echo "Running '{{ PackageInstallCommand }}'..."
 echo
 {{ PackageInstallCommand }}
 
 {{ if NpmRunBuildCommand | IsNotBlank }}
 echo
-echo "Running '{{ NpmRunBuildCommand }}' ..."
+echo "Running '{{ NpmRunBuildCommand }}'..."
 echo
 {{ NpmRunBuildCommand }}
 {{ end }}
 
 {{ if NpmRunBuildAzureCommand | IsNotBlank }}
 echo
-echo "Running '{{ NpmRunBuildAzureCommand }}' ..."
+echo "Running '{{ NpmRunBuildAzureCommand }}'..."
 echo
 {{ NpmRunBuildAzureCommand }}
 {{ end }}
 
+if [ "$ORYX_COPY_ONLY_PROD_MODULES_TO_OUTPUT" == "true" ]
+then
+	if [ "$SOURCE_DIR" != "$DESTINATION_DIR" ] && [ "$hasProductionOnlyDependencies" == "true" ]
```

Comment:
the condition for those `mv` commands to execute should be equivalent to the lines 17 + 19, right? If we rename the folders there, we should rename them back here

Reply:
We are renaming the folders here but I think I am not handling a case like:

**Scenario-1**
1. first build with copyonlyprodmodules=true
2. first build with copyonlyprodmodules=false (the source folder would still have the folders with the reserved names)

Will update the PR.